internal/config/config_integration_test.go

```diff
@@ -35,6 +35,12 @@ func TestConfigSetLogLevel(t *testing.T) {
 
 	err = c.Set("logLevel", "INVALID_VALUE")
 	assert.Error(t, err)
+
+	err = c.Set("loglevel", "Info")
+	assert.Error(t, err)
+
+	err = c.Set("Loglevel", "Debug")
+	assert.Error(t, err)
```

Comment:
What do you think about also including a couple tests for a `c.Get()` just to ensure that we're getting back what we expect also.  I think that would probably effectively test the round trip here.